index.bs

```diff
@@ -651,10 +651,10 @@ To <dfn>abort the request</dfn> |request|:
 </div>
 
 <div algorithm>
-To <dfn>signal to abort the request</dfn> |request|:
+To <dfn>signal to abort the request</dfn> |request| with |signal|:
 
 1. [=enqueue the following steps|Enqueue the steps=] to [=abort the request=] |request| to the [=lock task queue=].
-1. [=Reject=] |request|'s [=lock request/promise=] with an "{{AbortError}}" {{DOMException}}.
+1. [=Reject=] |request|'s [=lock request/promise=] with |signal|'s [=AbortSignal/abort reason=].
```

Comment:
I think it can't be undefined (https://github.com/whatwg/fetch/pull/1343#discussion_r746728681) but please correct me if I'm wrong.

Reply:
I can't remember if IDL argument processing is clever enough to treat an explicit `undefined` into "not given", for `AbortSignal.abort(undefined)` https://dom.spec.whatwg.org/#dom-abortsignal-abort and `controller.abort(undefined)` https://dom.spec.whatwg.org/#dom-abortcontroller-abort - but I'd agree that the intent is that abort reason is always a non-undefined value when aborted.

Reply:
IDL is that good fortunately.

Reply:
Hmmm, is the pattern of propagating the AbortSignal's reason out through rejections going to apply everywhere?

I'm not opposed, but I could see it being vaguely surprising that previously the only type of rejection was a DOMException, now it can be any value, e.g. `catch(reason => { console.log(reason.message); })` can no longer be trusted. 

Not a compatibility issue though since developers would need to start providing reasons themselves, so again I'm not opposed, just want to make sure this is being applied consistently.

Reply:
Yeah, that is the plan. https://dom.spec.whatwg.org/#abortcontroller-api-integration was updated to reflect this. And as per OP there's a list of specifications to update to take that into account.

cc @domenic